lib/spack/spack/binary_distribution.py

```diff
@@ -935,10 +935,19 @@ def build_tarball(spec, outdir, force=False, rel=False, unsigned=False,
     spackfile_path = os.path.join(
         cache_prefix, tarball_path_name(spec, '.spack'))
 
+    outdir = mirror.push_url
     remote_spackfile_path = url_util.join(
+        mirror.fetch_url, os.path.relpath(spackfile_path, tmpdir))
+    local_spackfile_path = url_util.join(
         outdir, os.path.relpath(spackfile_path, tmpdir))
 
     mkdirp(tarfile_dir)
+    if web_util.url_exists(local_spackfile_path):
+        if force:
+            web_util.remove_url(local_spackfile_path)
```

Comment:
I think it's weird to be possibly calling `remove_url` on what should be the same tarball twice.  If you remove it via the `local_spackfile_path` (the `push_url` version), it should also no longer be present to be removed in the block below.